tests/dialects/test_snowflake.py

```diff
@@ -1599,3 +1605,18 @@ def test_try_cast(self):
 
         expression = annotate_types(expression)
         self.assertEqual(expression.sql(dialect="snowflake"), "SELECT TRY_CAST(FOO() AS TEXT)")
+
+    def test_parse_alter_table(self):
+        """Assert that ALTER TABLE with UNIQUE column constraints can be parsed properly"""
+
+        # test parsing with single unique column
+        query = "ALTER TABLE authors ADD CONSTRAINT c1 UNIQUE (email)"
+        alter = parse_one(query, read="snowflake")
+        self.assertIsInstance(alter, exp.AlterTable)
+        self.assertEqual(query, alter.sql())
+
+        # test parsing with multiple unique columns
+        query = "ALTER TABLE authors ADD CONSTRAINT c1 UNIQUE (id, email)"
+        alter = parse_one(query, read="snowflake")
+        self.assertIsInstance(alter, exp.AlterTable)
+        self.assertEqual(query, alter.sql())
```

Comment:
I believe we lately have been avoiding adding tests like this one, right @tobymao? Can probably just have a couple of identity tests.

Other than that, one observation is that it's good practice to provide `dialect="snowflake"` in these `sql()` calls.